### DISCUSSION TITLE
I need to pass env to docker container I see it we have command kamal env push but it does not seems to push the file inside roles is always empty.

### DISCUSSION THREAD

**Discussion by ahsan-eazyops** [basecamp/kamal#563] [Q&A] [answered]
I need to pass env to docker container I see it we have command kamal env push but it does not seems to push the file inside roles is always empty.

**Comment by tangit86** [comment]
Check [here](https://kamal-deploy.org/docs/configuration)

Any env var that you'd need it to be passed to the containers, needs to be declared under 'env' section in your `deploy.yml` config (top level):

`env:
  clear:
    DATABASE_URL: mysql2://db1/hey_production/
    REDIS_URL: redis://redis1:6379/1
  secret:
    - DATABASE_PASSWORD
    - REDIS_PASSWORD
    `
same for 'accessories'  (under each accessory section)

**Comment by nickhammond** [comment, marked as answer]
@ahsan-eazyops It doesn't actually push the file into the container, it just runs the container with the `--env-file` option pointing at that .env file. The .env file is actually sitting on the server in a Kamal directory, if you look at the output from `kamal env push` you'll see where it stores it for that role + server. 

Docs: https://docs.docker.com/engine/reference/commandline/run/#env
